### PR TITLE
Unit test running parallel sim setup

### PR DESCRIPTION
**Pull request by sawenzel**

### CODE REVIEW

File: run/O2HitMerger.h [hunk 1]
```diff
@@ -135,6 +135,36 @@ class O2HitMerger : public FairMQDevice
...
+    index++;
+
+    auto itrackrefs = o2::Base::decodeTMessage<o2::dataformats::MCTruthContainer<o2::TrackReference>*>(data, index);
+    br = o2::Base::getOrMakeBranch(*mOutTree, "IndexedTrackRefs", &itrackrefs);
+    br->SetAddress(&itrackrefs);
+    br->Fill();
+    br->ResetAddress();
+    index++;
+  }
+
```
Comment (aphecetche) [inline run/O2HitMerger.h]: not exactly a good example of DRY as it's basically 3 times the same code (with different types). Can't the 3 call the same function ? Would be clearer.

Comment (sawenzel) [reply run/O2HitMerger.h]: Certainly can and it was on my list. Will improve it.

Comment (sawenzel) [reply run/O2HitMerger.h]: Done

**Comment by sawenzel** [comment]
The unit test doesn't seem to work yet (timeout). I have to split this PR.

**Pull request closed without merging**